### PR TITLE
BUILD-5391 setup pre commit validation at pr level

### PR DESCRIPTION
**Pull request by sebastienvermeille**
# BUILD-5391 setup pre commit validation at pr level

## Changes
* Enable pre-commit validation at PR level
* Update pre-commit-config so that it verify also github action and workflows (provides valuable inputs!)
* Fix issues so that any new contributor as of today do not have to fix earlier issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,13 +49,13 @@ jobs:
...
 
-Using the versioned semantic [tags](#Tags) is recommended for security and reliability.
+Using the versioned semantic [tags](#tags) is recommended for security and reliability.
 
 See [GitHub: Using tags for release management](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management)
 and [GitHub: Keeping your actions up to date with Dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/keeping-your-actions-up-to-date-with-dependabot)
 .
 
-For convenience, it is possible to use the [branches](#Branches) following the major releases.
+For convenience, it is possible to use the [branches](#branches) following the major releases.
```
Comment (julien-carsique-sonarsource) [inline README.md:58]: This change is not correct. The anchor name has a capital.

File: README.md [hunk 2]
```diff
@@ -49,13 +49,13 @@ jobs:
 
 ## Versioning
 
-Using the versioned semantic [tags](#Tags) is recommended for security and reliability.
+Using the versioned semantic [tags](#tags) is recommended for security and reliability.
```
Comment (julien-carsique-sonarsource) [inline README.md:52]: This change is not correct. The anchor name has a capital.

File: action.yml [hunk 3]
```diff
@@ -64,4 +61,3 @@ runs:
         file_glob: true
         file: "${{ inputs.filename }}?(.asc)"
         tag: ${{ github.ref }}
-        overwrite: true
```
Comment (julien-carsique-sonarsource) [inline action.yml:67]: Why the removal of this option?

File: README.md (hunk 1 above)
Comment (sebastienvermeille) [reply README.md:58]: @julien-carsique-sonarsource did you try? 

I did, and it actually works as expected; that issue was reported by the markdown linter.

File: README.md (hunk 2 above)
Comment (sebastienvermeille) [reply README.md:52]: @julien-carsique-sonarsource did you try?

I did, and it actually works as expected; that issue was reported by the markdown linter.

File: action.yml (hunk 3 above)
Comment (sebastienvermeille) [reply action.yml:67]: Good catch, ok I removed this hook which is currently not usable apparently:

```
    hooks:
      - id: check-github-actions
```

We can add it later, let's go baby steps it was not there initially.

Comment (sebastienvermeille) [reply action.yml]: Added here c1747fc6976954d626fa50c84bed06910a4b7211

**Review by julien-carsique-sonarsource — APPROVED** [review]

**Pull request merged**